### ISSUE TITLE
"Missing form label" accessibility error

### ISSUE THREAD

**Issue by ifrountas** [tomusborne/generatepress#573]
### Description
The function` generate_do_search_fields()` in the path `generatepress\inc\structure\search-modal.php` it makes inaccessible the label because:

1. Label is missing the for attribute
2. The input field doesn't have id in order to name the label.

### Steps to reproduce
1. Go to Customizing > Layout > Primary Navigation and check Enable navigation search modal
2. Just check the site with [Wave tool](https://wave.webaim.org/)

#### Actual behavior
... I just [made an override](https://capture.dropbox.com/yjeJof3vTNA1dDCw) to the search field in my child theme but I think it would be better to fixed at the theme level.

#### Expected behavior
...

---
